scripts/modules/b2b-account/account-address-search.js

```diff
@@ -0,0 +1,270 @@
+                    self.filterGrid(collection, self);
+                }, 400); 
+            }
+            });  
+            $('[data-mz-action="addEmailfilter"]').on('keyup input', function(e) {
+                e.preventDefault();                                                
+                clearTimeout(timeout);     
+                if(($("#searchEmail").val().length>=3) || ($("#searchEmail").val().length === 0))
+                {        
+                timeout = setTimeout(function () {
+                    self.filterGrid(collection, self);
+                }, 400); 
+            }
+            });  
+            $('[data-mz-action="addCountryfilter"]').on('keyup input', function(e) {
+                e.preventDefault();                                                
+                clearTimeout(timeout);       
+                if(($("#searchCountry").val().length>=3) || ($("#searchCountry").val().length === 0))
+                {      
+                timeout = setTimeout(function () {
+                    self.filterGrid(collection, self);
+                }, 400); 
+            }
+            }); 
+            $('[data-mz-action="addZipCodefilter"]').on('keyup input', function(e) {
+                e.preventDefault();                                                
+                clearTimeout(timeout);             
+                if(($("#searchZipCode").val()!=="") || ($("#searchZipCode").val().length === 0))
+                {
+                timeout = setTimeout(function () {
+                    self.filterGrid(collection, self);
+                }, 400); 
+            }
+            });      
+            $('[data-mz-action="clearSearch"]').on('click', function(e) {                 
+                 $("#searchAddress").val("");
+                 $("#searchCity").val("");
+                 $("#searchState").val("");
+                 $("#searchEmail").val("");
+                 $("#searchCountry").val("");
+                 $("#searchZipCode").val("") ;
+                 clearTimeout(timeout);             
+                 timeout = setTimeout(function () {
+                 self.filterGrid(collection, self);
+                 }, 400); 
+            });  
+                
+        },   
+        callbackForGridSelection: function(data) 
+        {
+            if(data !== undefined)
+            {
+                console.log(data.apiModel.data.accountId);
+            }
+        },
+       
+        filterGrid:function(collection, self)
+        {
+            var addressstring = "address.address1 cont";
+            var citystring = "address.cityortown cont";
+            var statestring = "address.stateorprovince cont";
+            var emailstring = "email cont";            
+            var countrystring = "address.countrycode cont";            
+            var zipcodestring = "address.postalorzipCode cont";
+            filterstring = "";
+            
+                if($("#searchAddress").val()!=="")
```

Comment:
You can move this if into createFilterString(). check this non empty string into createFilterString itself. rather than add if statement for individual filters.